src/game.cpp

```diff
@@ -5958,11 +5985,19 @@ int game::mon_info(WINDOW *w)
                 vec.push_back( critter.type );
             }
         } else if( p != nullptr ) {
-            if (p->attitude == NPCATT_KILL)
-                if (rl_dist( u.pos(), p->pos() ) <= iProxyDist) {
+            //Safemode NPC check
+
+            const int mondist = rl_dist( u.pos(), p->pos() );
```

Comment:
Looks like a repetition of the code above for the monster (only obvious difference is "human" instead of `critter.name()`), even the variable `mondist` (should be `npcdist`, shouldn't it?) - this figuratively screams to be put into a function/lambda.